### PR TITLE
[ARTEMIS-1536]: Incorrect Journal filesize calculation where specified size is lest that the block size when using AIO.

### PR DESCRIPTION
**Pull request by ehsavoie**
* If the specified file size is under the fs block size then the resulting file size is 0. Setting it to the block size in this case.

Jira: https://issues.apache.org/jira/browse/ARTEMIS-1536

### CODE REVIEW

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java [hunk 1]
```diff
@@ -163,13 +163,17 @@ protected void init(Configuration config, IOCriticalErrorListener criticalErrorL
...
       int fileSize = config.getJournalFileSize();
       // we need to correct the file size if its not a multiple of the alignement
-      int modulus = fileSize % journalFF.getAlignment();
-      if (modulus != 0) {
-         int difference = modulus;
-         int low = config.getJournalFileSize() - difference;
-         int high = low + journalFF.getAlignment();
-         fileSize = difference < journalFF.getAlignment() / 2 ? low : high;
-         ActiveMQServerLogger.LOGGER.invalidJournalFileSize(config.getJournalFileSize(), fileSize, journalFF.getAlignment());
+      if (fileSize <= journalFF.getAlignment()) {
```
Comment (franz1981) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: If `alignment` is a power of 2 (and it should be right?) you can use:
```
   public static long align(final long value, final long alignment) {
      return (value + (alignment - 1)) & ~(alignment - 1);
   }
```
That is into `BytesUtils::align`

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java [hunk 2]
```diff
@@ -163,13 +163,17 @@ protected void init(Configuration config, IOCriticalErrorListener criticalErrorL
...
+         fileSize = journalFF.getAlignment();
```
Comment (clebertsuconic) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: I would actually throw an exception here. I wouldn't expect a block size that low. you can only have the header on the journal and nothing else if you specify fileSize == alignment.

Comment (ehsavoie) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: There is already such an exception when creating the journal (java.lang.IllegalArgumentException: File size cannot be less than 1024 bytes
    Caused by: java.lang.IllegalArgumentException: File size cannot be less than 1024 bytes") . Would setting to high be acceptable ?

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: Lets not complicate things... but a unit test would be nice :)

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: I will just mere it.. but if you can add an unit test...

Comment (ehsavoie) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/JournalStorageManager.java]: Done

**Comment by clebertsuconic** [comment]
merged & cherry-picked into 2.6.x

**Pull request closed without merging**